server/controllers/notifications.js

```diff
@@ -0,0 +1,75 @@
+const Notifications = require("../models/Notifications");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /notifications
+// @desc get unread notifications for a user
+// @access private
+exports.getUnreadNotifications = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+
+  const notifications = await Notifications.find({ userId: id, read: false });
+
+  res.status(200).json({
+    success: {
+      notifications: notifications,
+    },
+  });
+});
+
+// @route GET /notifications/all
+// @desc get all notifications for a user
+// @access private
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+
+  const notifications = await Notifications.find({ userId: id });
+
+  res.status(200).json({
+    success: {
+      notifications: notifications,
+    },
+  });
+});
+
+// @route PUT /notifications/:id
+// @desc mark a notification as read
+// @access private
+exports.markNotificationAsRead = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const { notificationId } = req.body;
+
+  let update = await Notifications.findOneAndUpdate(
```

Comment:
On the client side - this will have a loop and multiple API requests per notification. Can be problematic if you think on a large scale if you have thousands of users all with unread notifications. It adds a lot to the amount of API requests we're taking on the server. 

I think a better approach would to take a batch of unread notifications and on the database do a batch operation on updating all their statuses. Reason for this - the DB is optimized for performing this operation. Plus, only one API request would need to be done passing a list of all the unread notifications (just the ids as well)